### PR TITLE
Add __builtin_fmaf16.

### PR DESCRIPTION
**Pull request by zahiraam**

### CODE REVIEW

**Comment by zahiraam** [comment]
@AaronBallman ping?

File: clang/include/clang/Basic/Builtins.td [hunk 1]
```diff
@@ -203,6 +203,12 @@ def FmaF16F128 : Builtin, F16F128MathTemplate {
   let Prototype = "T(T, T, T)";
 }
 
+def Fmaf16 : Builtin, F16F128MathTemplate {
```
Comment (AaronBallman) [inline clang/include/clang/Basic/Builtins.td:206]: Why is this needed when we have: https://github.com/llvm/llvm-project/blob/a3913a19185e3513a317aab792e85d5c50ef0242/clang/include/clang/Basic/Builtins.td#L201

That seems to already support fp16 for fma?

**Review by AaronBallman — APPROVED** [review]
Changes LGTM but please be sure to add a release note when landing so users know about the improvement.

**Comment by zahiraam** [comment]
> Changes LGTM but please be sure to add a release note when landing so users know about the improvement.

Thanks for the review.
